contracts/contracts/vault/OETHBaseZapper.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { IWETH9 } from "../interfaces/IWETH9.sol";
+import { IERC4626 } from "../../lib/openzeppelin/interfaces/IERC4626.sol";
+
+contract OETHBaseZapper {
+    IERC20 public immutable oethb;
+    IERC4626 public immutable woethb;
+    IVault public immutable vault;
+
+    IWETH9 public constant weth =
+        IWETH9(0x4200000000000000000000000000000000000006);
+    address private constant ETH_MARKER =
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    event Zap(address indexed minter, address indexed asset, uint256 amount);
+
+    constructor(
+        address _oethb,
+        address _woethb,
+        address _vault
+    ) {
+        oethb = IERC20(_oethb);
+        woethb = IERC4626(_woethb);
+        vault = IVault(_vault);
+
+        weth.approve(address(_vault), type(uint256).max);
+        IERC20(_oethb).approve(_woethb, type(uint256).max);
+    }
+
+    /**
+     * @dev Deposit ETH and receive OETH in return.
+     * Will verify that the user is sent 1:1 for ETH.
+     */
+    receive() external payable {
+        deposit();
+    }
+
+    /**
+     * @dev Deposit ETH and receive superOETHb in return
+     * Will verify that the user is sent 1:1 for ETH.
+     * @return Amount of OETH sent to user
+     */
+    function deposit() public payable returns (uint256) {
+        uint256 balance = address(this).balance;
```

Comment:
Since a `receive` function also triggers the `deposit` one this should only really happen when: 
- contract self destruct sends ETH to zapper
- ethereum POS validator sends beacon chain rewards to zapper. 

Both seem unlikely and I can't see the financial incentive.